hazelcast/src/main/java/com/hazelcast/client/impl/clientside/HazelcastClientInstanceImpl.java

```diff
@@ -329,9 +329,9 @@ private void startMetrics() {
         ThreadMetricSet.register(metricsRegistry);
         ClassLoadingMetricSet.register(metricsRegistry);
         FileMetricSet.register(metricsRegistry);
-        metricsRegistry.scanAndRegister(clientExtension.getMemoryStats(), "memory");
-        metricsRegistry.collectMetrics(clientExtension);
-        metricsRegistry.collectMetrics(executionService);
+        metricsRegistry.registerStaticMetrics(clientExtension.getMemoryStats(), "memory");
+        metricsRegistry.provideMetrics(clientExtension);
```

Comment:
I can't say I'm terribly excited about the name 'provide metrics'.

The caller will not provide metrics; it will collect the metrics that were provided.

Reply:
Ah...  this is the registration of metrics; not the collection of metrics.

Perhaps we should just call it something with register? I got confused.. so probably an indication of me either being stupid or the API not being that clear (of a combination of these 2).

Reply:
It's named `provideMetrics` since we call methods on `StaticMetricsProvider` instances that will register their metrics. There is an indirection. Also since this change metrics collection is the term used for iterating through the known and discovered metrics and collect (publish, render etc) them. I wanted to avoid the name clash with that too.